Identity 1.0.0 - Diataxis + Code Examples - iteration 1 (#1328)
iotaledger/iota-wiki: docs/build/identity.rs/1.0/docs/explanations/decentralized-identifiers.mdx

image: /img/Identity_icon.png
keywords:
  - public keys
  - Method Specification
  - Decentralized Identifiers
  - overview
  - DLT
---

# Decentralized Identifiers (DID)

:::info TL;DR

DIDs are unique identifiers that can be resolved to DID Documents containing public keys and URIs for identity verification and public information.

Adhering to the W3C's DID specifications, IOTA's implementation ensures interoperability and security within its ledger.

DIDs support self-sovereign identity, allowing identity owners to control their creation and destruction,
while facilitating encrypted communication.

:::

A DID is a unique identifier that can be resolved to a DID Document. This document contains data such as public keys, enabling the holder to prove ownership over their personal data, but also URIs that link to public information about the identity. DIDs are the fundamental building blocks of decentralized digital identity.
This implementation complies to the [DID specifications v1.0](https://www.w3.org/TR/did-core//) from the World Wide Web Consortium (W3C).

> In the IOTA Identity framework, we have implemented the DID standard according to the `iota` [DID Method Specification](/identity.rs/references/specifications/iota-did-method-spec/). Other implementations of DID on IOTA must follow the `iota` DID Method Specification if they also want to use the `iota` method name. It should not be necessary to re-implement the specification, if you can use library though.

Reviewer: ```suggestion
In the IOTA Identity framework, we have implemented the DID standard according to the `iota` [DID Method Specification](/identity.rs/references/specifications/iota-did-method-spec/). Other implementations of DID on IOTA must follow the `iota` DID Method Specification if they also want to use the `iota` method name. Libraries implementing the `iota` DID Method Specification are provided for [Rust](../getting-started/rust.mdx) and [WASM](../getting-started/wasm.mdx).
```

Keep in mind that, unfortunately, most of these methods are outdated and not maintained.

## DID Documents

The purpose of a DID is to help navigate to a DID Document,
which is a document containing more information regarding the identity subject.
This document contains data such as public keys, enabling the subject to prove ownership over their personal data,
but can contain additional information on how to interact with the subject.

The identifier contains all information to resolve a DID, providing the latest DID Document.
The first three characters `did` indicate that the DID standard from W3C must be used to resolve the identifier.
It is followed by a unique method name, in our case `iota`, to indicate that the IOTA method is used.

The IOTA method is a specific implementation following the [IOTA DID Method Specification](/identity.rs/references/specifications/iota-did-method-spec/).
This provides unique rules for the protocol to follow in order to manage a DID Document.
In our case, it describes how DID Documents are uploaded and queried to and from the IOTA ledger.

> Lastly, a DID also contains a set of random characters that are unique per identity.
> This makes the identity unique and makes sure every identity resolves to a single DID Document.

Reviewer: ```suggestion
Lastly, a DID also contains a set of random characters that are unique per identity and resolve to a single DID Document.
```

:::tip Requires basic knowledge of Asymmetric Encryption

The following sections require some basic knowledge of Asymmetric Encryption.
Please read or view some materials on the subject before continuing.

:::

> A DID Document mostly contains two important pieces of data: public keys and services.
> The public keys can be used to prove ownership over the identity,

Reviewer: ```suggestion
A DID Document mostly contains two important pieces of data: verification methods and services.

Verification methods contain public key material that can be used to prove ownership over the identity,
```

The public key can be used to verify that the identity subject signed the data and therefore controls the private key.
Ownership over the private keys, therefore, proves ownership over the identity.

This also means that it is very important to keep the private keys safe and secure.
In addition, the public keys allow users to send encrypted data to the identity that only the identity owner can decrypt.

:::caution

Never share your private keys, seeds, passphrases with anyone. Not even IOTA Foundation members.
This may lead to loss of funds or control over your own digital identity.

:::

Services are URIs that point to more information about the identity.
This could be something as simple as a website for an organizational identity.
These services are publicly available for all to read,
and should not contain Personal Identifiable Information (PII) in the case of human identities.

## Why Use DIDs?

DIDs allow any subject to have a unique identifier that they can prove ownership of,
and at the same time provide a way to send them encrypted messages.
The Identity is Self-Sovereign, meaning the subject is always in control;
whether it is creating, updating or destroying it.

> DIDs become more interesting when you combine them [Verifiable Credentials (VC)](verifiable-credentials.mdx).

Reviewer: ```suggestion
DIDs become more interesting when you combine them with [Verifiable Credentials (VC)](verifiable-credentials.mdx).
```
